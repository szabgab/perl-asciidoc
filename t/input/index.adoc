---
layout: documentation
title: Sample title
section: doc
---
ifdef::backend-html5[]
:description:
:author:
:email: address@googlegroups.com
:sectanchors:
:toc: left
:notitle:
endif::[]


Fee _text_
second line

More _text_  and an internal link: <<doc/developer#,Extend Asciidoc>>

== Title with 2 equal characters

Free text.

=== Title with 3 equal characters

More text
'''
++++
include::_partials/_footer.html[]
++++
