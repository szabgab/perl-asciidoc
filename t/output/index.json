{
   "content" : [
      {
         "cont" : [
            "Fee ",
            {
               "cont" : "text",
               "tag" : "b"
            },
            "",
            "second line"
         ],
         "tag" : "p"
      },
      {
         "cont" : [
            "More ",
            {
               "cont" : "text",
               "tag" : "b"
            },
            "  and an internal link: ",
            {
               "cont" : "Extend Asciidoc",
               "link" : "doc/developer#",
               "tag" : "a"
            },
            " another ",
            {
               "cont" : "Other Link",
               "link" : "../install",
               "tag" : "a"
            },
            " and more"
         ],
         "tag" : "p"
      },
      {
         "tag" : "h2",
         "text" : "Title with 2 equal characters"
      },
      {
         "cont" : [
            "Free text."
         ],
         "tag" : "p"
      },
      {
         "tag" : "h3",
         "text" : "Title with 3 equal characters"
      },
      {
         "cont" : [
            "More text"
         ],
         "tag" : "p"
      }
   ],
   "header" : {
      "layout" : "documentation",
      "section" : "doc",
      "title" : "Sample title"
   }
}
