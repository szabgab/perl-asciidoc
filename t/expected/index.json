{
   "ASCIIDOC" : {
      "AfterBody" : "'''\n++++\ninclude::_partials/_footer.html[]\n++++\n",
      "Body" : {
         "Page" : [
            {
               "PageBody" : {
                  "BodyPart" : [
                     {
                        "Paragraph" : "Free text. link:/local[Broken"
                     },
                     {
                        "Paragraph" : "Title] postfix."
                     },
                     {
                        "List" : {
                           "ListItem" : [
                              "* first list item",
                              "\n* second list item"
                           ]
                        }
                     },
                     {
                        "Comment" : "////\nsome comment\n\n3rd line of comment\n////"
                     },
                     {
                        "List" : ""
                     }
                  ]
               },
               "PageTitleLine" : {
                  "PageTitle" : "Title with 2 equal characters",
                  "PageTitleSize" : "=="
               }
            },
            {
               "PageBody" : {
                  "BodyPart" : [
                     {
                        "Source" : "[source,bash]\n----\nbrew install jenkins\n  verbatim text for code\n----"
                     },
                     {
                        "Source" : "[source, perl]\n----\nuse strict;\nuse warnings;\n----"
                     },
                     {
                        "Source" : "[source]\n----\n$ no language\n  provide\n  but it is still verbatim\n----"
                     },
                     {
                        "List" : ""
                     }
                  ]
               },
               "PageTitleLine" : {
                  "PageTitle" : "Title with 3 equal characters",
                  "PageTitleSize" : "==="
               }
            },
            {
               "PageBody" : {
                  "BodyPart" : [
                     {
                        "Special" : "[CAUTION]\n====\nThis text is to be handled as a paragraph of type caution.\n===="
                     },
                     {
                        "Paragraph" : "Regular paragraph."
                     },
                     {
                        "Special" : "[NOTE]\n====\nThis text is to be handled as a paragraph of type note.\n===="
                     },
                     {
                        "List" : ""
                     }
                  ]
               },
               "PageTitleLine" : {
                  "PageTitle" : "Special sections",
                  "PageTitleSize" : "==="
               }
            },
            {
               "PageBody" : {
                  "BodyPart" : [
                     {
                        "Paragraph" : "More text"
                     },
                     {
                        "List" : ""
                     }
                  ]
               },
               "PageTitleLine" : {
                  "PageTitle" : "End of page",
                  "PageTitleSize" : "==="
               }
            }
         ],
         "PageBody" : {
            "BodyPart" : [
               {
                  "Paragraph" : "Fee _text_"
               },
               {
                  "Paragraph" : "second line"
               },
               {
                  "Paragraph" : "More _text_  and an internal link: <<doc/developer#,Extend Asciidoc>> another link:../install[Other Link] and more"
               },
               {
                  "List" : ""
               }
            ]
         }
      },
      "Header" : {
         "Pair" : [
            {
               "Key" : "layout",
               "Value" : "documentation"
            },
            {
               "Key" : "title",
               "Value" : "Sample title"
            },
            {
               "Key" : "section",
               "Value" : "doc"
            }
         ]
      },
      "Ifdef" : "ifdef::backend-html5[]\n:description:\n:author:\n:email: address@googlegroups.com\n:sectanchors:\n:toc: left\n:notitle:\nendif::[]"
   }
}
